File: apps/api/app/schemas/audit.py
"""Mirror of packages/contracts/audit.schema.json."""

from pydantic import AwareDatetime, BaseModel, ConfigDict

from app.schemas.common import ActorRole, ActorType


class AuditRecord(BaseModel):
    """Append-only hash-chained accountability record."""

    model_config = ConfigDict(extra="forbid")

    audit_id: str
    ts: AwareDatetime
    actor_type: ActorType
    action: str
    entity_type: str
    hash_prev: str
    hash_self: str
    actor_id: str | None = None
    actor_role: ActorRole | None = None
    entity_id: str | None = None
    plant_id: str | None = None
    before: dict[str, object] | None = None
    after: dict[str, object] | None = None
    reason: str | None = None
    request_id: str | None = None
    session_id: str | None = None
    source_ip: str | None = None

File: apps/web/src/app/schemas/audit.ts
/** Mirror of packages/contracts/audit.schema.json */

import type { ActorRole, ActorType } from "./common";

export interface AuditRecord {
  audit_id: string;
  /** RFC3339 UTC */
  ts: string;
  actor_type: ActorType;
  action: string;
  entity_type: string;
  hash_prev: string;
  hash_self: string;
  actor_id?: string | null;
  actor_role?: ActorRole | null;
  entity_id?: string | null;
  plant_id?: string | null;
  before?: Record<string, unknown> | null;
  after?: Record<string, unknown> | null;
  reason?: string | null;
  request_id?: string | null;
  session_id?: string | null;
  source_ip?: string | null;
}